### DISCUSSION TITLE
Tangling the org file fails

### DISCUSSION THREAD

**Discussion by sonnius** [manna-harbour/miryoku#46] [Q&A] [answered]
tangling the org file fails

## while executing org-babel-tangle most output blocks are empty

I'm running this on MacOS

> emacs --version
GNU Emacs 27.1
org version 9.4

I've tried setting the python interpreter to:
   Python 2.7.16
and
   Python 3.8.5

Both versions seem to have similar issue

## Additional Context

happy to supply additional debug to solve this issue...

emacs messages:

executing Python code block (table-enums)...
Can’t guess python-indent-offset, using defaults: 2
Wrote /var/folders/p2/1vbnh94x4vz8777klk11xmmc0000gn/T/babel-FOK85z/ob-input-8qOOD2
Babel evaluation exited with code 1
""
Contacting host: raw.githubusercontent.com:443 [2 times]
executing Python code block (table-layout-taphold)...
Shell native completion is disabled, using fallback
""
executing Python code block (table-layout-taphold)...
"Traceback (most recent call last):
  File \"<stdin>\", line 11, in <module>
  File \"<string>\", line 30
SyntaxError: ’return’ outside function
"
Contacting host: raw.githubusercontent.com:443
executing Python code block (table-layout-taphold)...
"Traceback (most recent call last):
  File \"<stdin>\", line 11, in <module>
  File \"<string>\", line 30
SyntaxError: ’return’ outside function
"
executing Python code block (table-layout-taphold)...
"Traceback (most recent call last):
  File \"<stdin>\", line 11, in <module>
  File \"<string>\", line 30
SyntaxError: ’return’ outside function
"
Contacting host: raw.githubusercontent.com:443
executing Python code block (table-layout-taphold)...
"Traceback (most recent call last):
  File \"<stdin>\", line 11, in <module>
  File \"<string>\", line 30
SyntaxError: ’return’ outside function
"
executing Python code block (table-layout-taphold)...
"Traceback (most recent call last):
  File \"<stdin>\", line 11, in <module>
  File \"<string>\", line 30
SyntaxError: ’return’ outside function
"
executing Python code block (table-layout-half)...
Can’t guess python-indent-offset, using defaults: 2
Wrote /var/folders/p2/1vbnh94x4vz8777klk11xmmc0000gn/T/babel-FOK85z/ob-input-o2aknL
Babel evaluation exited with code 1
""
Contacting host: raw.githubusercontent.com:443
executing Python code block (table-layout-half)...
Can’t guess python-indent-offset, using defaults: 2
Wrote /var/folders/p2/1vbnh94x4vz8777klk11xmmc0000gn/T/babel-FOK85z/ob-input-gOfbsP
Babel evaluation exited with code 1
""
executing Python code block (table-layout-full)...
Can’t guess python-indent-offset, using defaults: 2
Wrote /var/folders/p2/1vbnh94x4vz8777klk11xmmc0000gn/T/babel-FOK85z/ob-input-iPHZyt
"  [MBO] = LAYOUT_miryoku(
    KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,
    KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,
    KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,
    KC_NP,   KC_NP,   KC_TRNS, KC_TRNS, KC_TRNS, KC_BTN1, KC_BTN3, KC_BTN2, KC_NP,   KC_NP
  )"
Contacting host: raw.githubusercontent.com:443
executing Python code block (table-layout-half)...
Can’t guess python-indent-offset, using defaults: 2
Wrote /var/folders/p2/1vbnh94x4vz8777klk11xmmc0000gn/T/babel-FOK85z/ob-input-GUZWoF
Babel evaluation exited with code 1
""
executing Python code block (table-layout-half)...
Can’t guess python-indent-offset, using defaults: 2
Wrote /var/folders/p2/1vbnh94x4vz8777klk11xmmc0000gn/T/babel-FOK85z/ob-input-I3GTYm
Babel evaluation exited with code 1
""
executing Python code block (table-layout-half)...
Can’t guess python-indent-offset, using defaults: 2
Wrote /var/folders/p2/1vbnh94x4vz8777klk11xmmc0000gn/T/babel-FOK85z/ob-input-YRluou
Babel evaluation exited with code 1
""
Contacting host: raw.githubusercontent.com:443
executing Python code block (table-layout-half)...
Can’t guess python-indent-offset, using defaults: 2
Wrote /var/folders/p2/1vbnh94x4vz8777klk11xmmc0000gn/T/babel-FOK85z/ob-input-Yi4kxG
Babel evaluation exited with code 1
""
Wrote //Users/rob/qmk_firmware/users/manna-harbour_miryoku/manna-harbour_miryoku.h
Wrote //Users/rob/qmk_firmware/users/manna-harbour_miryoku/manna-harbour_miryoku.c
Wrote //Users/rob/qmk_firmware/users/manna-harbour_miryoku/config.h
Wrote //Users/rob/qmk_firmware/layouts/community/ergodox/manna-harbour_miryoku/config.h
Wrote //Users/rob/qmk_firmware/layouts/community/ergodox/manna-harbour_miryoku/keymap.c
Wrote //Users/rob/qmk_firmware/layouts/community/ortho_4x12/manna-harbour_miryoku/config.h
Wrote //Users/rob/qmk_firmware/layouts/community/ortho_4x12/manna-harbour_miryoku/keymap.c
Wrote //Users/rob/qmk_firmware/keyboards/crkbd/keymaps/manna-harbour_miryoku/config.h
Wrote //Users/rob/qmk_firmware/keyboards/crkbd/keymaps/manna-harbour_miryoku/keymap.c
Wrote //Users/rob/qmk_firmware/users/manna-harbour_miryoku/rules.mk
Tangled 10 code blocks from miryoku.org

**Comment by manna-harbour** [comment, marked as answer]
Thanks for the report.

Please use the development branch, rather than what's in QMK master, as that requires a very old org version.

https://github.com/manna-harbour/qmk_firmware/blob/miryoku/users/manna-harbour_miryoku/miryoku.org

**Comment by sonnius** [comment]
thank you, getting the newest version solved the problem.

Thanks!
